modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -11,11 +11,45 @@
 
 enum { DIRBIT = 1, FILEBIT = 2 };
 
+static BYTE GetPIDLType(LPCITEMIDLIST pidl)
+{
+    return pidl && pidl->mkid.cb >= 3 ? (pidl->mkid.abID[0] & 0x7F) : 0;
+}
+
+struct FS95
+{
+    WORD cb;
+    BYTE type;
+    BYTE unknown;
+    UINT size;
+    WORD date, time;
+    WORD att;
+    CHAR name[ANYSIZE_ARRAY];
+
+    static BOOL IsFS(LPCITEMIDLIST p)
+    {
+        return (p && p->mkid.cb > 2) ? (p->mkid.abID[0] & 0x70) == 0x30 : FALSE;
+    }
+    static FS95* Validate(LPCITEMIDLIST p)
+    {
+        C_ASSERT(FIELD_OFFSET(FS95, name) == 14);
+        return p && p->mkid.cb > FIELD_OFFSET(FS95, name) && IsFS(p) ? (FS95*)p : NULL;
+    }
+};
+
 static int FileStruct_Att(LPCITEMIDLIST pidl)
```

Comment:
Why not make the helper just returning an `UINT`-typed value instead? (this would also avoid the ugly `UINT(1)` cast below. You could use `1UL << 31` also.)

Reply:
hmh then why not just replacing `UINT(1) << 31` by `-1` ?

Reply:
Because I want the WORD attributes to be zero so any & tests are false.